minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ConfigSchema.java

```diff
@@ -142,128 +125,10 @@ public ConfigSchema(Map map) {
-        if (remoteProcessingGroups != null) {
-            remoteInputPortIds.addAll(remoteProcessingGroups.stream().filter(r -> r.getInputPorts() != null)
-                    .flatMap(r -> r.getInputPorts().stream()).map(RemoteInputPortSchema::getId).collect(Collectors.toSet()));
-        }
-
-        Set<String> problematicDuplicateNames = new HashSet<>();
-        Set<String> missingProcessorNames = new HashSet<>();
-        // Set unset ids
-        connections.stream().filter(connection -> StringUtil.isNullOrEmpty(connection.getId())).forEachOrdered(connection -> connection.setId(getUniqueId(idMap, connection.getName())));
-
-        connections.stream().filter(connection -> StringUtil.isNullOrEmpty(connection.getSourceId())).forEach(connection -> {
-            String sourceName = connection.getSourceName();
-            if (remoteInputPortIds.contains(sourceName)) {
-                connection.setSourceId(sourceName);
-            } else {
-                if (duplicateProcessorNames.contains(sourceName)) {
-                    problematicDuplicateNames.add(sourceName);
-                }
-                String sourceId = processorNameToIdMap.get(sourceName);
-                if (StringUtil.isNullOrEmpty(sourceId)) {
-                    missingProcessorNames.add(sourceName);
-                } else {
-                    connection.setSourceId(sourceId);
-                }
-            }
-        });
-
-        connections.stream().filter(connection -> StringUtil.isNullOrEmpty(connection.getDestinationId()))
-                .forEach(connection -> {
-                    String destinationName = connection.getDestinationName();
-                    if (remoteInputPortIds.contains(destinationName)) {
-                        connection.setDestinationId(destinationName);
-                    } else {
-                        if (duplicateProcessorNames.contains(destinationName)) {
-                            problematicDuplicateNames.add(destinationName);
-                        }
-                        String destinationId = processorNameToIdMap.get(destinationName);
-                        if (StringUtil.isNullOrEmpty(destinationId)) {
-                            missingProcessorNames.add(destinationName);
-                        } else {
-                            connection.setDestinationId(destinationId);
-                        }
-                    }
-                });
-
-        if (problematicDuplicateNames.size() > 0) {
-            addValidationIssue(CANNOT_LOOK_UP_PROCESSOR_ID_FROM_PROCESSOR_NAME_DUE_TO_DUPLICATE_PROCESSOR_NAMES
-                    + problematicDuplicateNames.stream().collect(Collectors.joining(", ")));
-        }
-        if (missingProcessorNames.size() > 0) {
-            addValidationIssue(CONNECTIONS_REFER_TO_PROCESSOR_NAMES_THAT_DONT_EXIST + missingProcessorNames.stream().sorted().collect(Collectors.joining(", ")));
```

Comment:
I agree we should validate that the connections refer to existing ids, will add that